src/core/Form/Textarea/Textarea.tsx

```diff
@@ -1,52 +1,162 @@
 import { withSuomifiDefaultProps } from '../../theme/utils';
-import classnames from 'classnames';
+import { HtmlTextarea, HtmlTextareaProps, HtmlDiv } from '../../../reset';
+import { AutoId } from '../../../utils/AutoId';
+import { LabelText } from '../LabelText/LabelText';
+import { HintText } from '../HintText/HintText';
+import { StatusText } from '../StatusText/StatusText';
+import { InputStatus } from '../types';
 
 const baseClassName = 'fi-textarea';
-
 const textareaClassNames = {
+  textareaContainer: `${baseClassName}_textarea-element-container`,
+  textarea: `${baseClassName}_textarea`,
   disabled: `${baseClassName}--disabled`,
   error: `${baseClassName}--error`,
 };
 
-export interface TextareaProps extends CompTextareaProps, TokensProp {}
+type TextareaStatus = Exclude<InputStatus, 'success'>;
 
-const StyledTextarea = styled(
-  ({ tokens, ...passProps }: TextareaProps & InternalTokensProp) => (
-    <CompTextarea {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
-`;
+export interface TextareaProps extends HtmlTextareaProps, TokensProp {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on textarea text change */
+  onChange?: (event: ChangeEvent<HTMLTextAreaElement>) => void;
+  /** To execute on textarea text onBlur */
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: TextareaStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  /** Optional text that is shown after labelText. Will be wrapped in parentheses. */
+  optionalText?: string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Input name */
+  name?: string;
+}
 
-export class Textarea extends Component<TextareaProps> {
+class BaseTextarea extends Component<TextareaProps> {
   render() {
     const {
-      children,
+      id,
+      className,
       disabled = false,
+      children,
+      onClick,
+      labelMode,
+      labelText,
+      hintText,
       status,
-      className,
+      statusText,
+      visualPlaceholder,
+      resize: dismissResize,
```

Comment:
This pattern of renaming to indicate discarding a prop is used inconsistently in this repo. My proposal would be not to use it since it's not a commonly used best practice and is easy to forget.

Reply:
As discussed, this will be just destructured without renaming, so it is now:
`resize,`